Framework/Algorithms/src/XrayAbsorptionCorrection.cpp

```diff
@@ -0,0 +1,234 @@
+                                                        Direction::Output),
+                  "The name to use for the output workspace.");
+
+  auto positiveDouble = std::make_shared<Kernel::BoundedValidator<double>>();
+  positiveDouble->setLower(0);
+  declareProperty("DetectorAngle", DEFAULT_ANGLE, positiveDouble,
+                  "Angle between beam and Detector."
+                  "Range of normal values for detectors are :  "
+                  "Ge1 : 90-180 , Ge2 : 270-360 , Ge3 : 0 - 90 , Ge4 "
+                  ": 180 -270.",
+                  Direction::Input);
+
+  declareProperty("DetectorDistance", DEFAULT_DETECTOR_DISTANCE, positiveDouble,
+                  "Distance between detector and sample.", Direction::Input);
+}
+
+std::map<std::string, std::string> XrayAbsorptionCorrection::validateInputs() {
+  MatrixWorkspace_sptr inputWS = getProperty("InputWorkspace");
+  MatrixWorkspace_sptr muonProfile = getProperty("MuonImplantationProfile");
+  std::map<std::string, std::string> issues;
+  if (!inputWS->sample().getShape().hasValidShape()) {
+    issues["InputWorkspace"] = "Input workspace does not have a Sample";
+  } else {
+    auto material = inputWS->sample().getShape().material();
+    if (!material.hasValidxRayAttenuationOverride()) {
+      issues["InputWorkspace"] =
+          "Input workspace does not have a Xray Attenuation profile";
+    } else {
+      if (muonProfile->getNumberHistograms() != 1) {
+        issues["MuonImplantationProfile"] =
+            "Muon Implantation profile must have only one spectrum";
+      }
+    }
+  }
+  return issues;
+}
+
+/**
+ * Converts angles in degrees to angles in radians.
+ * @param degrees double of angle in degrees
+ * @return A double of angle in radians
+ */
+double XrayAbsorptionCorrection::degreesToRadians(double degrees) {
+  double radians = M_PI * (degrees / 180);
+  return radians;
+}
+
+/**
+ * Calculates the position of the detector.
+ * @param detectorAngle double of angle of detector
+ * @param detectorDistance distance betweeen sample and detector
+ * @return V3D object of position of detector
+ */
+Kernel::V3D
+XrayAbsorptionCorrection::calculateDetectorPos(double detectorAngle,
+                                               double detectorDistance) {
+  detectorDistance = detectorDistance / 100;
+  double x = detectorDistance / std::tan(degreesToRadians(detectorAngle));
+  if (detectorAngle > 180) {
+    detectorDistance = -detectorDistance;
+  }
+  Kernel::V3D detectorPos = {x, detectorDistance, 0.};
+  return detectorPos;
+}
+
+/**
+ * normalise moun intensity to 1.
+ * @param muonIntensity A MantidVec which contains the intensity of muons as a
+ * function  of depth
+ * @return A vector of doubles which contains the normalised muon intensity
+ */
+std::vector<double> XrayAbsorptionCorrection::normaliseMuonIntensity(
+    MantidVec const &muonIntensity) {
+  double sum_of_elems{0};
+  for (double elem : muonIntensity) {
```

Comment:
if you make a vector then you can use std::accumulate https://en.cppreference.com/w/cpp/algorithm/accumulate